parlai/core/torch_agent.py

```diff
@@ -666,7 +786,27 @@ def add_cmdline_args(
             help='add person tokens to history. adds __p1__ in front of input '
             'text and __p2__ in front of past labels when available or '
             'past utterances generated by the model. these are added to '
-            'the dictionary during initialization.',
+            'the dictionary during initialization. If --speaker-field is '
+            "specified and is present in the observation, `observation['speaker_field']` "
+            'will get used instead of __p1__.',
+        )
+        agent.add_argument(
+            '-spkr',
+            '--speaker-field',
+            type='nonestr',
+            default=None,
+            help="field in the observation to track indicating the id of the speaker "
+            "over the course of the episode (defaults to None); `observation['speaker_field']` "
+            "overrides `__p1__` in `--person-tokens` if field is not None and is present in the observation.",
+        )
```

Comment:
so we have the `id` field already in Messages - what do we think about just defaulting to that?